code-challenges.test.js

```diff
@@ -28,12 +35,93 @@
+
+// const fibb = (num) => {
+//   finArr = []
+// }
+
+// Ok so the formula is x(n) = x(n-1) + x(n-2)
+// I am trying to figure out how to come up with the next index so the new index should be x(n)... or maybe the array at an index? a(i)...
+// So the formula can then be written a(i)=a(i-1)+a(i-2). This is starting to look or read like a for loop. I think... I guess I can try it.
+
+// for(let i = 0; i < array.length; i++){
+//   return a(i) = a(i-1)+a(i-2)
+// }
+// The answer I will push into the empty array so
+
+// for(let i = 0; i < array.length; i++){
+//   return finArr.push(a(i) = a(i-1)+a(i-2))
+// }
+
+// Ok, I will add it to the function and see what happens
+
+// const fibb = (num) => {
+//   finArr = []
+//   for(let i = 0; i < array.length; i++){
+//     return finArr.push(a(i) = a(i-1)+a(i-2))
+//   }  return finArr
+// }
+// console.log(fibb(6))
+
+// Welp failed miserably. Also I am reading the directions again and it needs to take in a number greater than 2 so I am most likely going to need a conditional for that. Also, I am just realizing my for loop doesn't have an array to iterate over so there's also that. Fun.
+
+// const fibb = (num) => {
+  // In order for the formula to start it needs an array. so maybe it starts with one number just to initiate the formula?
+  // Also I forgot to switch the ()from the equation to []. *insert eyeroll here*
+  // arr = [0]
+  // finArr = []
+  // if(num > 2){
+  //   for(let i=0; i<= arr.length; i++){
+  //     arr[i] = arr[i-1] + arr[i-2]
+  //     finArr.push(arr[i])
+      // I think I can adjust the array length returned by assigning it the number in the parameter so I will do that...
+      // Nope, didn't work..
+    // }
+    // finArr.length = num - 1
+
+//     return finArr
+//   }
+// }
+
+
+
+// So I am fairly certain I got the fibb sequence to work. It 1. kept going after 6 indexes and 2. stopped working after 6 indexes. Hmmm. off to google I go...
+// Ok, so first thing is first. In my research I found that (i) needed to be set to  <= the length of the number I am passing in so it can run to that length.
+// Ok, that worked except I am missing one 1 and after position 6 getting NAN...
+// After some pen and paper work (so fun) of running through my formula there I realized:
+  // 1. i needs to start at index 2 in order for it to not be a negative index number so I will add another number in and...just missing a 1
+  // 2. I am missing a 1 because the way the math works the first number pushed into the new array is the second 1 in the sequence since I had to start at index 2. So I added 1 to my finArr... and it passed!
+
+  const fibb = (num) => {
+    arr = [0,1]
+    finArr = [1]
+    if(num > 2){
+      for(let i=2; i<= num; i++){
+        arr[i] = arr[i-1] + arr[i-2]
+        finArr.push(arr[i])
+        }
+        return finArr
+      }
+    }
+    // woo, that was a doozy
```

Comment:
I am actually muttering "good job" under my breath as you figured this out. Great Work!